File: client/src/api/game.ts
import fetchApi from "@/packages/fetchApi.ts";
import type {Currency} from "@/api/common.ts";

export enum JoinType {
  ANYONE = 'anyone',
  FRIENDS = 'friends',
  LINK = 'link'
}

export interface CreateGameInput {
  currency_id: number
  bet: number
  winning_points: number
  join_type: JoinType
}

export interface Game {
  id: number
  code: string
  bet: number
  winning_points: number
  link: string
  currency: Currency
}

export const createGame = async (input: CreateGameInput): Promise<Game> => {
  const data = await fetchApi.post('/games', input)

  return data.data;
}

File: server/responses/game.go
package responses

import "app/models"

type GameResource struct {
	ID            uint             `json:"id"`
	Code          string           `json:"code"`
	Bet           uint             `json:"bet"`
	WinningPoints uint             `json:"winning_points"`
	Link          string           `json:"link"`
	Currency      CurrencyResource `json:"currency"`
}

func NewGameResource(game models.Game) GameResource {
	return GameResource{
		ID:            game.ID,
		Code:          game.Code,
		Currency:      NewCurrencyResource(game.Currency),
		Bet:           game.Bet,
		WinningPoints: game.WinningPoints,
		Link:          game.Code,
	}
}
